app/controllers/CyaCreateRecordController.scala

```diff
@@ -63,9 +68,31 @@ class CyaCreateRecordController @Inject() (
   }
 
   def onSubmit(): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
-    auditService
-      .auditFinishCreateGoodsRecord(request.eori, request.affinityGroup, request.userAnswers)
-      .map(_ => Redirect(routes.CreateRecordSuccessController.onPageLoad().url))
+    CreateGoodsRecordRequest.build(request.userAnswers, request.eori) match {
+      case Right(model) =>
+        goodsRecordConnector
```

Comment:
is it possible to put this in a for comprehension? easier to process

Reply:
done but see comment